### PR TITLE
Reidweber/ch1917/create network calls for clubhub core

### PR DESCRIPTION
**Pull request by Reidweb1**
Added missing services, interfaces, and base client class.

### CODE REVIEW

File: src/client/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import Axios from 'axios'
+import * as axios from 'axios'
+
+/**
+ * `ClubHubClient` is a thin wrapper around the Axios library. 
+ */
+export default class ClubHubClient {
+
+    /**
+     * The underling Axios instance.
```
Comment (JohnDaly) [inline src/client/index.ts]:
```suggestion
     * The underlying Axios instance. 
```

File: src/client/index.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * Return a successful request, no need to perform any actions.
+     */
+    private responseHandler = (response: any): any => {
+        return response
+    }
+
+    /**
+     * Check the error to see if it's a `401`. Otherwise throw the
```
Comment (JohnDaly) [inline src/client/index.ts]: Old comment, or is that what we want this function to do?

File: src/service/calendar.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ */
+export default class CalendarService {
+
+  /**
+   * The `ClubHubClient` instance for the service. 
+   */
+  public client: ClubHubClient
+
+  constructor(client: ClubHubClient) {
+    this.client = client
```
Comment (sgruse) [inline src/service/calendar.ts]: Looks like you have different indentation for spacing. Can we go with tab indentation for this project

File: src/models/calendar.ts [hunk 4]
```diff
@@ -4,8 +4,26 @@ import { Types } from 'mongoose'
 // Sub Models.
 import Location from './subModels/shared/location'
 
+// Required local Namespace.
+import Event from './event'
+
 namespace Calendar {
 
+	// --------------------------------
+	// Server Responses
```
Comment (sgruse) [inline src/models/calendar.ts]: I think we should put the server responses in a different Namespace so we can keep model definitions and route definitions separate. Also adding comment breaks (//--------------) that describe the type of server response. For example not all Calendar server responses are of the 'paged' response type.

File: src/service/event.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  public client: ClubHubClient
+
+  constructor(client: ClubHubClient) {
+    this.client = client
+  }
+
+  /**
+   * `GET` all the Events for the given calendar groupID.
+   */
+  public getEvents = async (groupID?: string, limit?: string, offset?: string, start?: string, end?: string): Promise<Event.Response> => {
```
Comment (JohnDaly) [inline src/service/event.ts]: We might want to think about changing the parameters to be an object, which has all of these existing parameters as optional keys. It makes it way easier to deal with everything being optional, and easier to add new fields as we go

File: src/service/member.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  public client: ClubHubClient
+
+  constructor(client: ClubHubClient) {
+    this.client = client
+  }
+
+  /**
+   * `GET` all the Members for a given club Id.
+   */
+  public getMembers = async (clubID: string, limit?: string, offset?: string, start?: string, end?: string): Promise<User.Response> => {
```
Comment (JohnDaly) [inline src/service/member.ts]: See earlier comment about making parameters into an object

File: src/service/message.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  public client: ClubHubClient
+
+  constructor(client: ClubHubClient) {
+    this.client = client
+  }
+
+  /**
+   * `GET` all the Messages for a given delivery type.
+   */
+  public getMessages = async (type?: Message.DeliveryType, limit?: string, offset?: string, start?: string, end?: string): Promise<Message.Model> => {
```
Comment (JohnDaly) [inline src/service/message.ts]: See earlier comment about parameters

File: src/service/post.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  public client: ClubHubClient
+
+  constructor(client: ClubHubClient) {
+    this.client = client
+  }
+
+  /**
+   * `GET` posts for the given parameters.
+   */
+  public getPosts = async (limit?: string, offset?: string, start?: string, end?: string): Promise<Post.Response> => {
```
Comment (JohnDaly) [inline src/service/post.ts]: See earlier comment about parameters

File: src/models/request.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  export interface Event extends GetQuery {
+    groupID: string
+  }
+
+  export interface Member extends GetQuery {
+    clubID: string
+  }
+
+  export interface Message extends GetQuery {
+    type: Message.DeliveryType
```
Comment (JohnDaly) [inline src/models/request.ts]: `type` should be optional

File: src/models/request.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+  export interface GetQuery {
+    limit?: string
+    offset?: string
+    start?: string
+    end?: string
+  }
+
+  export interface Event extends GetQuery {
+    groupID: string
```
Comment (JohnDaly) [inline src/models/request.ts]: `groupID` should probably be optional

**Pull request merged**